### ISSUE TITLE
Release 1.21.0 – 1 February 2021

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#2948]
Context
- Retention retention retention
- Working towards items on the roadmap, primarily a smoother experience on the core analytics side

Support hero: Eric

Eric / Tim
- Insight improvements #2877 
- Funnel trends #1269
- Show period in progress retention #2833,
- Improve date selection for retention #2831 
- CH bug retention #2794
- Indicate that a query is cached and allow to update query #2931
- Sentry bug squashing
- Setup slack log bot

Paolo/Karl
- Fix & release weekly email #1938 #1983
- Move signup to React #2734
- ~Design new onboarding flow #2822 **🔍 Focus**~
- Implement above  **🔍 Focus**
- Smart/proactive analytics alerts #2773 **🤞 Stretch**
- ~Implementing new sessions filters UI (waiting on design)~
- Solving mystery reliability issue #2927
- Measuring rageclicks, rolling feature out to everyone

Marius / Michael
- Take plugins out of beta
- Continue work on getting plugins running on EE (mostly a bunch of tests to write) PostHog/plugin-server#34
- ~~Plugins on GitLab and GitLab private repos (asked by users) #2940 #2941~~
- ~~Plugin Ordering & design changes to the interface PostHog/plugin-server#102 #2939~~
- Improve plugin server reloads PostHog/plugin-server#59
- ~~Upgrade plugins inside the app PostHog/plugin-server#103~~
- ~~Github Star Sync plugin (clean up and put in repo)~~
- ~~API access in plugins (easier way to share personal API tokens and call e.g. annotations/create)~~
- Expand scheduling functions (runEvery15Minutes) PostHog/plugin-server#68
- Maxmind plugin --> set the country/city/etc on the Person as well
- Plugins on app #2898 and PostHog/plugin-server#34.
- Postgres ingestion to the plugin server
- There's some threading optimization to be made in the plugin server

James G
- Setup a cluster using clickhouse-operator for managing a clickhouse cluster
- Specifically do this on GCP to test GPC volume mounting and performance for netdata's use case
- Extend our current helm chart to include kafka
- Extend our current helm chart to include zookeeper to support kafka and clickhouse clusters
- Assist with continue work on getting plugins running on EE (mostly a bunch of tests to write) PostHog/plugin-server#34
- Continue with optimizations around how we store data in clickhouse to see if we can get better performance out of it
- Act as performance support for SuprDaily now that we have the major performance enhancements done
- Probably touch this during helm chart work (Helm Charts: you can't set the SITE_URL variable via the chart PostHog/charts#29)
- Documentation - We really need to document our new setup and some of the learnings that have been discovered in setting up a sharded and distributed clickhouse cluster. We will need this for onboarding new engineers and I'm sure the community would love the blog post versions.
  - current architecture
  - document all learnings from clickhouse

**Comment by paolodamico** [comment]
### E
1. #2857. Key missing piece of data instrumentation to help understand retention (and overall product usage).
1. Strong core analytics experience. Weed out as much of #2877 as possible. Problems in the core experience can greatly degrade the user's perspective and easily lead to churning. From https://github.com/PostHog/retention/issues/7, we know we greatly need to improve this. From within that, here's what _I_ would prioritize (please challenge):
2.     - #2817 (has been reported a few times by users)
    - #2817 (has been reported a few times by users)

I think where I can have most impact is on:

**Comment by paolodamico** [comment]
My thoughts,
1. #2857. Key missing piece of data instrumentation to help understand retention (and overall product usage).
1. Strong core analytics experience. Weed out as much of #2877 as possible. Problems in the core experience can greatly degrade the user's perspective and easily lead to churning. From https://github.com/PostHog/retention/issues/7, we know we greatly need to improve this. From within that, here's what _I_ would prioritize (please challenge):
    - Retention graph improvements (dogfooding): #2833, #2831, #2794.
    - #1269 (has been requested by users multiple times; basic functionality).
    - #2817 (has been reported a few times by users)
1. Really nail down the **onboarding experience** (#2822). We know from https://github.com/PostHog/retention/issues/3, **this is where we have the biggest drop today in terms of retention**. I would love to pair up with someone on this.
1. Related to the above, finish #2734 as it is part of the onboarding experience (albeit a simplified version). This is also geared towards improving sign ups & retention of internally referred users.
1. Fix & release **weekly email** (#1938, #1983, ...). We saw [positive results](https://posthog.slack.com/archives/C018RGRS10S/p1609796905007800) with the no-event-ingestion campaign. This could be another driver of engagement, worth dusting it off to try out this strategy now on the engagement side.

I'd like to work myself (as I think where I can have most impact) on: 3, 4 & 5.

In addition to the above, we just shipped a bunch of improvements on our own internal instrumentation because we were lacking visibility in some key functionality of the product (e.g. should we improve the toolbar?). As we wait for this new data to come in (to better understand where we need to focus next in our retention journey), I think it's worth exploring these two new ideas (in case anyone feels like taking them). Worth mentioning that exploring brand new functionality is particularly helpful, as we're exploring more avenues and don't require more data upfront.
- #2773 (proactive alerting). _Rationale:_ This is geared towards increasing engagement in our product. You'll keep coming back to PostHog if we alert you when things don't look right (plus it can help potentially expand use cases). This could in theory be quite powerful as it creates a more organic loop to come back to PH (same as you would jump on PagerDuty if you get an alert that something doesn't look right). It has also been requested by users.
- #2880 (product interviews). _Rationale:_ Won't repeat what's already documented in the issue, but in addition to that, this can be a feature that greatly drives collaboration within PostHog, increasing interaction between team members, which in turn can be a huge driver of engagement and therefore retention.


### Supporting analyses
1. Do users who are joined by multiple team members retain better?
1. Competitors / overlapping products deep dive.
1. Keep reaching out to schedule for customer feedback calls (including churned users).

**Comment by macobo** [comment]
There's some loose ends from last sprint:

1. Implementing new sessions filters UI (waiting on design)
2. Solving mystery reliability issue https://github.com/PostHog/posthog/issues/2927
3. Measuring rageclicks, rolling feature out to everyone

There's other low-hanging fruit for sessions after this (player UX, tagging support, allowing to filter by whether session has been viewed), but would love getting some qualitative/quantitative feedback first.

In other words - I am mostly a free agent! Where can I help?

wink-wink to James, Paolo, core analytics.

**Comment by mariusandra** [comment]
**Plugins:**
- Continue work on getting plugins running on EE (mostly a bunch of tests to write) https://github.com/PostHog/plugin-server/pull/34
- Plugins on GitLab and GitLab private repos (asked by users) https://github.com/PostHog/posthog/issues/2940 https://github.com/PostHog/posthog/issues/2941
- Plugin Ordering & design changes to the interface https://github.com/PostHog/plugin-server/issues/102 https://github.com/PostHog/posthog/pull/2939
- Improve plugin server reloads https://github.com/PostHog/plugin-server/issues/59
- Upgrade plugins inside the app https://github.com/PostHog/plugin-server/issues/103
- Github Star Sync plugin (clean up and put in repo)
- API access in plugins (easier way to share personal API tokens and call e.g. annotations/create)
- Expand scheduling functions (runEvery15Minutes) https://github.com/PostHog/plugin-server/issues/68
- Maxmind plugin --> set the country/city/etc on the Person as well

**Misc:**
- Support & Sentry: here's an idea. Instead of one support hero we could have two: one "support sheikh" and one "error emir". I could stay on top of the support issues, but that plus coding plus meetings plus sentry was just a bit too much. 
- TypeScript: it's painful to see the vanilla-JS code in frontend. I still want to help fix this and improve the quality of code. Some errors I saw this week came from this.
- Helm Charts: you can't set the SITE_URL variable via the chart https://github.com/PostHog/charts/issues/29

**Comment by Twixes** [comment]
Well, I had some things listed on plugins, but Marius beat me to it by a minute. Anyway, we discussed the direction of plugins today, so in addition to the above I'll just list a few personal highlights:

1. The new ingestion is working really nicely! But the last bit we want before moving our production pipeline to it is some solid tests, same as the whole suite Django runs now. The goal is to deploy early after these 2 PRs get merged with feedback addressed: #2898 and PostHog/plugin-server#34.
1. I'll add Postgres ingestion to the plugin server, since that's actually a trivial task now!
1. There's some threading optimization to be made in the plugin server.
1. Further improvements of the plugin development experience, like revamping https://github.com/PostHog/helloworldplugin into a nicer starting point with a tutorial.

Also, want to get webhook/REST hook tests in, as that seems to be a bit wonky from user reports. There may possibly be some other various tending to code as well, as had been in the past release cycle.

**Comment by EDsCODE** [comment]
Generally continuing analytics improvements.

#2877 almost done
Follow up with all the points in Paolo's 2nd point and #2228 and #2697
add info about when queries have been cached

**Comment by fuziontech** [comment]
The 'epic' is basically automate all of the things. Now that we have clickhouse setup for ourselves in a relatively bespoke fashion we can start spending some cycles automating a process to deploy our infrastructure for customers. We can use the learnings from setting up our own cluster and bake that into this automation. The automation will be a helm chart unless someone has a better idea.
- Setup a cluster using `clickhouse-operator` for managing a clickhouse cluster
- Specifically do this on GCP to test GPC volume mounting and performance for netdata's use case
- Extend our current helm chart to include kafka
- Extend our current helm chart to include zookeeper to support kafka and clickhouse clusters

Other work that needs to be done:
- Assist with continue work on getting plugins running on EE (mostly a bunch of tests to write) PostHog/plugin-server#34
- Continue with optimizations around how we store data in clickhouse to see if we can get better performance out of it
- Act as performance support for SuprDaily now that we have the major performance enhancements done
- Probably touch this during helm chart work (Helm Charts: you can't set the SITE_URL variable via the chart PostHog/charts#29)
- Documentation - We really need to document our new setup and some of the learnings that have been discovered in setting up a sharded and distributed clickhouse cluster. We will need this for onboarding new engineers and I'm sure the community would love the blog post versions.

**Comment by paolodamico** [comment]
This release has been shipped, closing.
